projects/core/components/calendar/calendar.component.ts

```diff
@@ -30,8 +30,18 @@ import {TuiMarkerHandler} from '@taiga-ui/core/types';
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TuiCalendarComponent implements TuiWithOptionalMinMax<TuiDay> {
-    @Input()
-    @tuiDefaultProp()
+    @Input('month')
+    set monthSetter(month: TuiMonth) {
+        const {computedMinViewedMonth, computedMaxViewedMonth} = this;
+
+        if (month.monthBefore(computedMinViewedMonth)) {
```

Comment:
I think it would look clearer if we moved that into separate function returning `TuiMonth`

Reply:
Smth like this?
```
 @Input('month')
    set monthSetter(month: TuiMonth) {
        this.month = this.getComputedMonth(month);
    }
    month = TuiMonth.currentLocal();
...
private getComputedMonth(month: TuiMonth): TuiMonth {
        const {computedMinViewedMonth, computedMaxViewedMonth} = this;

        if (month.monthBefore(computedMinViewedMonth)) {
            return computedMinViewedMonth;
        } else if (month.monthAfter(computedMaxViewedMonth)) {
            return computedMaxViewedMonth;
        } else {
            return month;
        }
    }
```

Reply:
Right, except with returns you would not need `else if`, just 2 separate `if` and `return` at the very end with no `else` so it would look better :)

Reply:
@waterplea hmm tests error out using that approach. i think we could also add a getter so all look classy?
```
@Input()
    set month(month: TuiMonth) {
        this._month = this.getComputedMonth(month);
    }
    get month() {
        return this._month;
    }
    _month = TuiMonth.currentLocal();

```

Reply:
I don't think we need a getter. We basically have property month and a setter just to make sure we get a valid value.

Reply:
yeah @waterplea but without the above getter these tests fail:
```
  it('if set month is less than viewed, it takes computed min viewed', () => {
        const month = TuiMonth.currentLocal();
        const monthAfter = month.append({month: 1});

        component.minViewedMonth = monthAfter;
        component.month = month;

        expect(component.month).toBe(monthAfter);
    });

    it('if set month is more than max, it takes max viewed', () => {
        component.max = TuiDay.currentLocal().append({month: -1});
        component.month = TuiDay.currentLocal();
        expect(component.month).toBe(component.max);
    });
```
Looks like .month doesn't access the setter, but the property itself.

Reply:
Yes, if you do it imperatively with code, change it out monthSetter =. Only angular templates know about input aliases

Reply:
ok @waterplea though now we have 2 ways of accessing the same value in the component: `monthSetter` and `month` which hits readability so IMO getter with `_month` could be preferable .

Reply:
Sure, that's reasonable. Make it the way you want. Underscore private props just tend to annoy me for some reason )

Reply:
@waterplea when we clamp our `month` in the setter we depend on other input values like `maxViewedMonth` , `max`, which means that we depend on the order in which the input bindings were set in the template as you now. So i don't think this cool , we could use ngOnCHanges because it doesn't look component's binding will be updated often or just remove the clamping overall.

Reply:
@waterplea removed clamping, i think this one can be merged if everything's ok